### ISSUE TITLE
Steam snap unusable

### ISSUE THREAD

**Issue by UnrealisticDino** [canonical/steam-snap#286]
**Describe the bug**
Zero games work. After pressing play it will process shaders, say running for a second, then the game stops.

**To Reproduce**
The steam snap has slowly gotten worse after the nvidia 535 driver update and now no games work.
1. Go to any game
2. Click on play
3. Scroll down to '...'
4. See steam try to launch the game but then quit before anything happens

**Expected behavior**
Games that used to work should still work. Games that gave errors should give the same errors.

**Screenshots**

**Environment**

`snap run steam.report`

```
os_release: 
    name:               "Ubuntu"
    version:            "22.04.2 LTS (Jammy Jellyfish)"
snap_info: 
    steam_revision:     146
    snapd_revision:     19457
lspci: 
    08:00.0:            NVIDIA Corporation GP104 [GeForce GTX 1070] (rev a1)
glxinfo: 
    gpu:                NVIDIA GeForce GTX 1070/PCIe/SSE2
    gpu_version:        4.6.0 NVIDIA 535.54.03
lscpu: 
    model_name:         AMD Ryzen 7 3800X 8-Core Processor
xdg_current_desktop:    ubuntu:GNOME
desktop_session:        ubuntu
```

**Logs**
Add any errors/warnings output to the terminal (if applicable).
Run Steam from the terminal with `snap run steam` to easily see output.
```
BuildCompleteAppOverviewChange: 977 apps
RegisterForAppOverview 1: 34ms
RegisterForAppOverview 2: 35ms
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed
ExecCommandLine: "'/home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/steam' '-no-cef-sandbox'"

(steam:20343): GLib-GIO-CRITICAL **: g_settings_schema_source_lookup: assertion `source != NULL' failed
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed
GameAction [AppID 314710, ActionID 1] : LaunchApp changed task to ProcessingInstallScript with ""
pressure-vessel-wrap[20842]: W: "snap/steam/146/graphics/usr/share/libdrm" is unlikely to appear in "/run/host"
pressure-vessel-wrap[20842]: W: "snap/steam/146/graphics/usr/share/libdrm" is unlikely to appear in "/run/host"
pressure-vessel-wrap[20842]: W: "var/lib/snapd/hostfs/usr/share/drirc.d" is unlikely to appear in "/run/host"
pressure-vessel-wrap[20842]: W: "snap/steam/146/graphics/usr/share/drirc.d" is unlikely to appear in "/run/host"
pressure-vessel-wrap[20842]: W: "var/lib/snapd/hostfs/usr/share/nvidia" is unlikely to appear in "/run/host"
pressure-vessel-wrap[20842]: W: "var/lib/snapd/hostfs/usr/share/nvidia" is unlikely to appear in "/run/host"
Proton: Error while copying to "/mnt/fast1/SteamLibrary/steamapps/compatdata/314710/pfx/drive_c/windows/system32/d3d12core.dll": No such file or directory
Proton: Error while copying to "/mnt/fast1/SteamLibrary/steamapps/compatdata/314710/pfx/drive_c/windows/syswow64/d3d12core.dll": No such file or directory
GameAction [AppID 314710, ActionID 1] : LaunchApp changed task to SynchronizingCloud with ""
GameAction [AppID 314710, ActionID 1] : LaunchApp changed task to SynchronizingControllerConfig with ""
GameAction [AppID 314710, ActionID 1] : LaunchApp changed task to SiteLicenseSeatCheckout with ""
GameAction [AppID 314710, ActionID 1] : LaunchApp changed task to CreatingProcess with ""
GameAction [AppID 314710, ActionID 1] : LaunchApp waiting for user response to CreatingProcess ""
GameAction [AppID 314710, ActionID 1] : LaunchApp continues with user response "CreatingProcess"
/bin/sh\0-c\0/home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=314710 -- /home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/juan/snap/steam/common/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun -- '/home/juan/snap/steam/common/.local/share/Steam/steamapps/common/Proton 8.0'/proton waitforexitandrun  '/mnt/fast1/SteamLibrary/steamapps/common/Mighty No. 9/Binaries/Win32/MN9Game.exe' -nohomedir -seekfreeloadingpcconsole > /home/juan/game_mightynative 2>&1\0
Game process added : AppID 314710 "/home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=314710 -- /home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/juan/snap/steam/common/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun -- '/home/juan/snap/steam/common/.local/share/Steam/steamapps/common/Proton 8.0'/proton waitforexitandrun  '/mnt/fast1/SteamLibrary/steamapps/common/Mighty No. 9/Binaries/Win32/MN9Game.exe' -nohomedir -seekfreeloadingpcconsole > /home/juan/game_mightynative 2>&1", ProcID 20914, IP 0.0.0.0:0
chdir "/mnt/fast1/SteamLibrary/steamapps/common/Mighty No. 9/binaries/Win32/"
ERROR: ld.so: object '/home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
flock /usr/share/fonts/truetype/liberation2/LiberationSans-Italic.ttf LOCK_SH failed. errno = 13flock /usr/share/fonts/truetype/liberation2/LiberationSans-Bold.ttf LOCK_SH failed. errno = 13flock /usr/share/fonts/truetype/liberation2/LiberationSans-Regular.ttf LOCK_SH failed. errno = 13flock /usr/share/fonts/truetype/liberation2/LiberationSans-Regular.ttf LOCK_SH failed. errno = 13flock /usr/share/fonts/truetype/liberation2/LiberationSans-Bold.ttf LOCK_SH failed. errno = 13GameAction [AppID 314710, ActionID 1] : LaunchApp changed task to WaitingGameWindow with ""

(steam:20343): GLib-GIO-CRITICAL **: g_settings_schema_source_lookup: assertion `source != NULL' failed
GameAction [AppID 314710, ActionID 1] : LaunchApp changed task to Completed with ""

(steam:20343): GLib-GIO-CRITICAL **: g_settings_schema_source_lookup: assertion `source != NULL' failed
Game process removed: AppID 314710 "/home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=314710 -- /home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/juan/snap/steam/common/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun -- '/home/juan/snap/steam/common/.local/share/Steam/steamapps/common/Proton 8.0'/proton waitforexitandrun  '/mnt/fast1/SteamLibrary/steamapps/common/Mighty No. 9/Binaries/Win32/MN9Game.exe' -nohomedir -seekfreeloadingpcconsole > /home/juan/game_mightynative 2>&1", ProcID 20914 
ThreadGetProcessExitCode: no such process 20989
ThreadGetProcessExitCode: no such process 20917
ThreadGetProcessExitCode: no such process 20916
ThreadGetProcessExitCode: no such process 20915
Uploaded AppInterfaceStats to Steam

(steam:20343): GLib-GIO-CRITICAL **: g_settings_schema_source_lookup: assertion `source != NULL' failed

(steam:20343): GLib-GIO-CRITICAL **: g_settings_schema_source_lookup: assertion `source != NULL' failed
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed
GameAction [AppID 314710, ActionID 2] : LaunchApp changed task to ProcessingInstallScript with ""
pressure-vessel-wrap[21012]: W: "snap/steam/146/graphics/usr/share/libdrm" is unlikely to appear in "/run/host"
pressure-vessel-wrap[21012]: W: "snap/steam/146/graphics/usr/share/libdrm" is unlikely to appear in "/run/host"
pressure-vessel-wrap[21012]: W: "var/lib/snapd/hostfs/usr/share/drirc.d" is unlikely to appear in "/run/host"
pressure-vessel-wrap[21012]: W: "snap/steam/146/graphics/usr/share/drirc.d" is unlikely to appear in "/run/host"
pressure-vessel-wrap[21012]: W: "var/lib/snapd/hostfs/usr/share/nvidia" is unlikely to appear in "/run/host"
pressure-vessel-wrap[21012]: W: "var/lib/snapd/hostfs/usr/share/nvidia" is unlikely to appear in "/run/host"
Proton: Error while copying to "/mnt/fast1/SteamLibrary/steamapps/compatdata/314710/pfx/drive_c/windows/system32/d3d12core.dll": No such file or directory
Proton: Error while copying to "/mnt/fast1/SteamLibrary/steamapps/compatdata/314710/pfx/drive_c/windows/syswow64/d3d12core.dll": No such file or directory
GameAction [AppID 314710, ActionID 2] : LaunchApp changed task to SynchronizingCloud with ""
GameAction [AppID 314710, ActionID 2] : LaunchApp changed task to SynchronizingControllerConfig with ""
GameAction [AppID 314710, ActionID 2] : LaunchApp changed task to SiteLicenseSeatCheckout with ""
GameAction [AppID 314710, ActionID 2] : LaunchApp changed task to CreatingProcess with ""
GameAction [AppID 314710, ActionID 2] : LaunchApp waiting for user response to CreatingProcess ""
GameAction [AppID 314710, ActionID 2] : LaunchApp continues with user response "CreatingProcess"
/bin/sh\0-c\0/home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=314710 -- /home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/juan/snap/steam/common/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun -- '/home/juan/snap/steam/common/.local/share/Steam/steamapps/common/Proton 8.0'/proton waitforexitandrun  '/mnt/fast1/SteamLibrary/steamapps/common/Mighty No. 9/Binaries/Win32/MN9Game.exe' -nohomedir -seekfreeloadingpcconsole > /home/juan/game_mightynative 2>&1\0
Game process added : AppID 314710 "/home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=314710 -- /home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/juan/snap/steam/common/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun -- '/home/juan/snap/steam/common/.local/share/Steam/steamapps/common/Proton 8.0'/proton waitforexitandrun  '/mnt/fast1/SteamLibrary/steamapps/common/Mighty No. 9/Binaries/Win32/MN9Game.exe' -nohomedir -seekfreeloadingpcconsole > /home/juan/game_mightynative 2>&1", ProcID 21084, IP 0.0.0.0:0
chdir "/mnt/fast1/SteamLibrary/steamapps/common/Mighty No. 9/binaries/Win32/"
ERROR: ld.so: object '/home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
GameAction [AppID 314710, ActionID 2] : LaunchApp changed task to WaitingGameWindow with ""

(steam:20343): GLib-GIO-CRITICAL **: g_settings_schema_source_lookup: assertion `source != NULL' failed
GameAction [AppID 314710, ActionID 2] : LaunchApp changed task to Completed with ""

(steam:20343): GLib-GIO-CRITICAL **: g_settings_schema_source_lookup: assertion `source != NULL' failed

(steam:20343): GLib-GIO-CRITICAL **: g_settings_schema_source_lookup: assertion `source != NULL' failed
Game process removed: AppID 314710 "/home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=314710 -- /home/juan/snap/steam/common/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/juan/snap/steam/common/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun -- '/home/juan/snap/steam/common/.local/share/Steam/steamapps/common/Proton 8.0'/proton waitforexitandrun  '/mnt/fast1/SteamLibrary/steamapps/common/Mighty No. 9/Binaries/Win32/MN9Game.exe' -nohomedir -seekfreeloadingpcconsole > /home/juan/game_mightynative 2>&1", ProcID 21084 
ThreadGetProcessExitCode: no such process 21159
ThreadGetProcessExitCode: no such process 21087
ThreadGetProcessExitCode: no such process 21086
ThreadGetProcessExitCode: no such process 21085
Uploaded AppInterfaceStats to Steam

(steam:20343): GLib-GIO-CRITICAL **: g_settings_schema_source_lookup: assertion `source != NULL' failed
```

**Additional Context**
```
[ 4394.143304] kauditd_printk_skb: 280 callbacks suppressed
[ 4394.143307] audit: type=1400 audit(1689445676.402:141483): apparmor="DENIED" operation="open" class="file" profile="snap.steam.steam" name="/proc/20007/environ" pid=20343 comm="steam" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
```

**Comment by ashuntu** [comment]
I can't reproduce any bad behavior on my machine, I have NVIDIA 525. I personally have never had much luck using the bleeding edge NVIDIA drivers, but since you tried 525 I'm not sure what the issue is. Although, I know at one point only NVIDIA 515 would work for me, but 525 works at present.

It may be worth trying different [gaming-graphics-core22 (stable) branches](https://github.com/canonical/steam-snap/wiki/FAQ#how-do-i-use-a-different-mesagraphics-version), but I can't promise much there.

@ZoopOTheGoop, I know you have NVIDIA, can you reproduce anything?

**Comment by UnrealisticDino** [comment]
I had to use sudo snap remove --purge steam to uninstall steam. 1.0.0.7.6 latest/edge seems to work but 1.0.0.76 latest/stable and 1.0.0.75 latest/beta didn't work. this did start after the nvidia 535 driver update but I have no idea why it broke and why it slowly stopped working.

**Comment by kenvandine** [comment]
I had a similar issue on my hybrid graphics system (Quadro T2000).  When I installed Lunar I got the 525 driver, and had the issue and downgrading to 515 worked.  Others couldn't reproduce the issue on 525.  I recently got upgraded from 515 to 535 and I ran into this issue again.  And now the 515 driver isn't available anymore.  I'm going to try downgrading to the 470 to see if it helps.

**Comment by kenvandine** [comment]
This is due to lack of 32 nvidia driver on the host system.  You need to enable i386 support and install the appropriate 

```
sudo dpkg --add-architecture i386 && sudo apt update
```
Then install the appropriate i386 driver package.  For example, the 535 driver is libnvidia-gl-535:i386 

```
sudo apt install libnvidia-gl-535:i386 
```

We need to detect the lack of this at start time and guide users to enable this.

**Comment by UnrealisticDino** [comment]
I used sudo snap remove --purge steam to uninstall steam 1.0.0.76 latest/edge then went back to 1.0.0.76 latest/stable and it's working now without doing sudo dpkg --add-architecture i386 && sudo apt update  or sudo apt install libnvidia-gl-535:i386. Steam did remember what proton version I forced in the game properties but still required a restart when changing it in the settings. If snap purge doesn't completely remove everything then maybe some file got corrupted and was only replaced by switching channels? Yesterday I had to use the terminal because I couldn't uninstall steam using the software center. Maybe saving logs from that would have been useful.

**Comment by UnrealisticDino** [comment]
It's starting to break again. Certain games no longer work, some are fine, some have performance issues.

**Comment by ashuntu** [comment]
Could you try https://github.com/canonical/steam-snap/issues/286#issuecomment-1640570498 for completeness sake? On my machine it doesn't seem to matter, but it seems to fix issues for @kenvandine

**Comment by UnrealisticDino** [comment]
Sorry about that. I should have included that I did try those but only after it started to break again but before making the comment. I did just reboot and it seems to be working fine again. I'll aim to include more relevant information in the future.

**Comment by UnrealisticDino** [comment]
I finally got around to doing a clean install, had the same issues, https://github.com/canonical/steam-snap/issues/286#issuecomment-1640570498 fixed it.

**Comment by ashuntu** [comment]
> I finally got around to doing a clean install, had the same issues, [#286 (comment)](https://github.com/canonical/steam-snap/issues/286#issuecomment-1640570498) fixed it.

If the problems you were experiencing were completely resolved, please close this issue.

Otherwise, it may warrant further digging.

**Comment by 4ndr3aR** [comment]
I can confirm that Steam 1.0.0.78 via snap is unusable. I think it has serious read/write access issues, I see that it manages to download Proton versions but it doesn't create pfx subfolders in compatdata/<appid>. Tried extensively with Elite Dangerous (any Proton version) and a few other small Windows games to confirm my theories. What's extremely strange is that writing  PROTON_LOG=1 %command% in launch options doesn't produce any log at all. No steam-<appid>.log file in $HOME or anywhere else. Changing PROTON_LOG_DIR in user_settings.py to /tmp is the only option to write something, and with something I mean a zero-byte file in /tmp/snap-private-tmp/snap.steam/tmp

**Comment by UnrealisticDino** [comment]
> I can confirm that Steam 1.0.0.78 via snap is unusable. I think it has serious read/write access issues, I see that it manages to download Proton versions but it doesn't create pfx subfolders in compatdata/. Tried extensively with Elite Dangerous (any Proton version) and a few other small Windows games to confirm my theories. What's extremely strange is that writing PROTON_LOG=1 %command% in launch options doesn't produce any log at all. No steam-.log file in $HOME or anywhere else. Changing PROTON_LOG_DIR in user_settings.py to /tmp is the only option to write something, and with something I mean a zero-byte file in /tmp/snap-private-tmp/snap.steam/tmp

The snap has stopped working for me twice and the way I fixed it was by purging the snap then reinstalling. Uninstalling via the software center did not work and would get stuck at about 80%. I haven't been able to replicate it but I think when it broke for me it had something to do with uninstalling Bloons 6.

**Comment by UnrealisticDino** [comment]
I am going to leave this issue closed though since it seems like you and hardBSDk are experiencing bugs that are not related to the issues I was having.

**Comment by 4ndr3aR** [comment]
I'm sorry, I spent the afternoon yesterday fighting with Steam and Proton, and that was mostly a rant out of frustration. Anyway I can confirm my initial thought: my problem looks like a serious permissions issue, Steam-snap seems to be able to write to half of the locations it should be entitled to write to. Yesterday night I finally removed the Steam snap and installed again via apt. Elite Dangerous worked mostly out of the box (I just had to choose Proton 7.0 probably because of my old Nvidia GTX 650). Hope this can someone in the future.

**Comment by ashuntu** [comment]
@4ndr3aR if you're having issues that others are not (and that no issue already exists for), it may be worthwhile to open a new issue if you feel inclined. If it's actually a permissions issue, the denied paths would likely appear in `sudo dmesg` and would be useful to note in the issue if you open one. My only surface level hints at anything permission related would be to make sure you have everything connected (`snap connections steam` lists them).

**Comment by lc-nyovchev** [comment]
@4ndr3aR @ashuntu I am almost certain that the issue we are all having is this one: https://github.com/canonical/steam-snap/issues/311

It's rather easy to reproduce.

- first you install `steam` via `snap`
- then you play some games, which download different `proton` versions depending on your config
- then, if you update your `nvidia` drivers and reboot, `steam` will still boot with no problem, but the installed `proton` versions would fail finding the proper drivers (they'd be searching in the old dirs), so you won't be able to start any game
- reinstalling the `steam` `snap` and then reinstalling `proton` via that makes your games work again (because the paths are now correct)

**Comment by UnrealisticDino** [comment]
> @4ndr3aR @ashuntu I am almost certain that the issue we are all having is a duplicate of this one: #311
> 
> It's rather easy to reproduce.
> 
>     * First you install `steam` via `snap`
> 
>     * Then you play some games, which download different `proton` versions depending on your config
> 
>     * Then, if you update your `nvidia` drivers and reboot, `steam` will still boot with no problem, but the installed `proton` versions would fail finding the proper drivers (they'd be searching in the old dirs), so you won't be able to start any game
> 
>     * Reinstalling the `steam` `snap` and then reinstalling `proton` via that makes your games work again (because the paths are now correct)

You might be right. It was working two weeks ago when I made a report running
```
os_release:
    name:               "Ubuntu"
    version:            "22.04.3 LTS (Jammy Jellyfish)"
snap_info:
    steam_revision:     171
    snapd_revision:     20290
lspci:
    08:00.0:            NVIDIA Corporation GP104 [GeForce GTX 1070] (rev a1)
glxinfo:
    gpu:                NVIDIA GeForce GTX 1070/PCIe/SSE2
    gpu_version:        4.6.0 NVIDIA 535.129.03
lscpu:
    model_name:         AMD Ryzen 7 3800X 8-Core Processor
xdg_current_desktop:    ubuntu:GNOME
desktop_session:        ubuntu
```
But is now broken again
```
os_release:
    name:               "Ubuntu"
    version:            "22.04.3 LTS (Jammy Jellyfish)"
snap_info:
    steam_revision:     171
    snapd_revision:     20671
lspci:
    08:00.0:            NVIDIA Corporation GP104 [GeForce GTX 1070] (rev a1)
glxinfo:
    gpu:                NVIDIA GeForce GTX 1070/PCIe/SSE2
    gpu_version:        4.6.0 NVIDIA 535.146.02
lscpu:
    model_name:         AMD Ryzen 7 3800X 8-Core Processor
xdg_current_desktop:    ubuntu:GNOME
desktop_session:        ubuntu

```

**Comment by D1E60E** [comment]
```i386-linux-gnu-capsule-capture-libs: warning: Dependencies of libEGL_nvidia.so.0 not found, ignoring: Missing dependencies: Could not find "libnvidia-glsi.so.535.154.05" in LD_LIBRARY_PATH "/snap/steam/171/graphics/lib/i386-linux-gnu:/snap/steam/171/graphics/usr/lib:/snap/steam/171/usr/lib/i386-linux-gnu:/snap/steam/171/lib/i386-linux-gnu```

This was my error and then I snap remove --purged the snap version 1.0.0.78 of the steam package

Then I installed apt version which gave me 1.0.0.74
This fixed it !

**Comment by ashuntu** [comment]
This problem is being tracked in https://github.com/canonical/steam-snap/issues/358 and https://github.com/canonical/steam-snap/issues/311.

**Comment by gazhay** [comment]
> @4ndr3aR @ashuntu I am almost certain that the issue we are all having is a duplicate of this one: #311
> 
> It's rather easy to reproduce.
> 
> * First you install `steam` via `snap`
> * Then you play some games, which download different `proton` versions depending on your config
> * Then, if you update your `nvidia` drivers and reboot, `steam` will still boot with no problem, but the installed `proton` versions would fail finding the proper drivers (they'd be searching in the old dirs), so you won't be able to start any game
> * Reinstalling the `steam` `snap` and then reinstalling `proton` via that makes your games work again (because the paths are now correct)

Just had this issue and confirm this solution does not work. 
After many graphics driver changes (and reboots) and multiple snap reinstalls (including redownloading all the games, adding accounts, etc) what was working fine 3 days ago no longer works. 

Logs are all but useless as they show 
things like 
```ERROR: ld.so: object '/home/gaz/snap/steam/common/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/gaz/snap/steam/common/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
```
or 
```
(steam:18683): GLib-GIO-CRITICAL **: g_settings_schema_source_lookup: assertion `source != NULL' failed
```

But all are variously described elsewhere as "red herrings". 

Rolled back as far as nvidia-525 but still nothing. Like I said was all happy just days ago.
